torch/csrc/jit/passes/graph_fuser.cpp

```diff
@@ -40,18 +89,35 @@ struct GraphFuser {
   bool isCuda(Node * node) {
     return node->type()->expect<TensorType>()->device() != -1;
   }
-  // TODO: the fusion compiler needs to know how to handle 'alpha'
-  // and other attributes in code generation for us to be able to fuse them
-  // then it is safe to remove the !hasSpecialAlpha check
-  bool hasSpecialAlpha(Node * node) {
-    if(!node->hasAttribute(kalpha))
+  // TODO: the fusion compiler has a lot of float-specific codegen
+  // so for now we only consider nodes that operate on floating point numbers
+  bool hasFloatType(Node * node) {
+    if(!node->hasType()) {
+      return false;
+    }
+    if(auto tt = node->type()->cast<TensorType>()) {
+      return tt->scalarType() != at::kFloat;
+    } else {
       return false;
-    return at::Scalar(node->t(kalpha)).toDouble() != 1;
+    }
+  }
+  bool hasFloatIO(Node * node) {
```

Comment:
This is more accurately named allFloatIO, since 'has' implies "has a floating point input/output" but that's not what is being tested here.